domain_config_ui/domain_config_ui.services.yml

```diff
@@ -0,0 +1,16 @@
+services:
+  domain_config_ui.manager:
+    class: Drupal\domain_config_ui\DomainConfigUIManager
+    arguments: ['@config.storage', '@domain.loader', '@language_manager']
```

Comment:
I haven't had a chance to really dive into this code. The part that bothers me is the introduction of config handling to the DomainNegotiator. That class/service should not really be Config aware. 

I have no problem making getDomainConfigName public, but I haven't worked through the pro-con.

In general:

* The core Domain module and its classes should handle domain registration and negotiation. That's all.
* Domain Config should handle storage and retrieval of config overrides.
* Domain Config UI should put a form in from of the read/write aspects of Domain Config.

Put another way, we should be able to programatically write an override through Domain Config.